### PR TITLE
BR-22 Write logic for bean name resolving

### PR DESCRIPTION
**Pull request by Laskovka**
### Issue/Task

Implement logic for bean name resolving

### Description

Write logic for bean name resolving

### Testing

Added/updated

### Related Reviews

-

### Screenshots

-

### CODE REVIEW

**Review by JJJazl — APPROVED** [review]

File: src/main/java/svydovets/util/BeanNameResolver.java [hunk 1]
```diff
@@ -2,12 +2,42 @@
...
+    //create javadoc for description
+    public static String resolveBeanName(Method initMethod) {
+        if (initMethod.isAnnotationPresent(Bean.class)) {
+            var beanValue = initMethod.getAnnotation(Bean.class).value();
+            return beanValue.isEmpty() ?
+                initMethod.getName() :
+                beanValue;
+        }
+        //throw exception if not bean annotated
+        throw new RuntimeException();
```
Comment (Pan14ek) [inline src/main/java/svydovets/util/BeanNameResolver.java:37]: Let's create a specific exception for this case

File: src/main/java/svydovets/util/BeanNameResolver.java [hunk 2]
```diff
@@ -2,12 +2,42 @@
...
     }
 
-    public static String resolveBeanNameByBeanInitMethod(Method initMethod) {
-        return null;
+    private static String setFirstCharacterToLowercase(String beanName) {
```
[the remark this answers is not in the document]
Comment (Laskovka) [reply src/main/java/svydovets/util/BeanNameResolver.java:40]: Discussed. It's okay.

File: src/test/java/com/bobocode/svydovets/util/BeanNameResolverTest.java [hunk 3]
```diff
@@ -23,25 +21,25 @@ public class BeanNameResolverTest {
     @DisplayName("")
     public void test() {
```
[the remark this answers is not in the document]
Comment (Laskovka) [reply src/test/java/com/bobocode/svydovets/util/BeanNameResolverTest.java:20]: Will fix it in scope of another ticket.

File: src/test/java/com/bobocode/svydovets/util/BeanNameResolverTest.java [hunk 4]
```diff
@@ -23,25 +21,25 @@ public class BeanNameResolverTest {
...
         String beanName = MessageService.class.getAnnotation(Component.class).value();
-        assertThat(beanName).isEqualTo(BeanNameResolver.resolveBeanNameByBeanType(MessageService.class));
+        assertThat(beanName).isEqualTo(BeanNameResolver.resolveBeanName(MessageService.class));
     }
 
     @Test
     @DisplayName("")
```
[the remark this answers is not in the document]
Comment (Laskovka) [reply src/test/java/com/bobocode/svydovets/util/BeanNameResolverTest.java:26]: The same as above

File: src/main/java/svydovets/util/BeanNameResolver.java [hunk 5]
```diff
@@ -2,12 +2,42 @@
...
+                return beanValue;
+            }
+        }
+        if (beanClass.isAnnotationPresent(Component.class)) {
+            var beanValue = beanClass.getAnnotation(Component.class).value();
+            if (!beanValue.isEmpty()) {
+                return beanValue;
+            }
+        }
+        return setFirstCharacterToLowercase(beanClass.getSimpleName());
```
Comment (Pan14ek) [inline src/main/java/svydovets/util/BeanNameResolver.java:25]: We can rename it to `toLoweCaseFirstCharacter`. I think it's okay to have the name with `to`. You can check the String class -> https://docs.oracle.com/javase/8/docs/api/java/lang/String.html

**Review by Pan14ek — COMMENTED** [review]
PTAL

**Review by JJJazl — APPROVED** [review]

File: src/main/java/svydovets/util/BeanNameResolver.java [hunk 6]
```diff
@@ -2,12 +2,42 @@
...
+    //create javadoc for description
+    public static String resolveBeanName(Class<?> beanClass) {
+        if (beanClass.isAnnotationPresent(Configuration.class)) {
+            var beanValue = beanClass.getAnnotation(Configuration.class).value();
+            if (!beanValue.isEmpty()) {
+                return beanValue;
+            }
+        }
+        if (beanClass.isAnnotationPresent(Component.class)) {
+            var beanValue = beanClass.getAnnotation(Component.class).value();
```
Comment (Pan14ek) [inline src/main/java/svydovets/util/BeanNameResolver.java:20]: Please add the `value` method to the `Component` annotation.

**Pull request merged**